drivers/i2c/Kconfig

```diff
@@ -218,7 +220,7 @@ config I2C_1
 config I2C_1_NAME
 	string "Port 1 device name"
 	default "I2C_1"
-	depends on I2C_1
+	depends on I2C_1 && !HAS_DTS_I2C
 
 config I2C_1_DEFAULT_CFG
```

Comment:
I'm not sure if we have already decided to go definitely for the latter approach but if we did we need to add 
depends on !HAS_DTS_I2C
here to exclude this config option for DTS based I2C drivers. Also, we need to ensure that the driver initialization function does not call i2c_configure(). 

If we remove I2C_1_DEFAULT_CFG from Kconfig.defconfig but not from here and the driver initialization we will attempt to configure the driver with invalid parameters at boot, this should result in an error.